process-engine/bpmn-studio
# 🐛 :bug: Check For XML File Signature Before Opening Diagrams

## What did you change?

**Changes**:

- Add diagram validator service fluent API.
- Extract single diagram opening code into a single function.
- Only allow files with XML signature to be opened.

Closes #560

## How can others test the changes?

- Start BPMN Studio as electron app
- Open single diagram via solution explorer icon
- Open diagram via double click feature

If the diagram is already opened a message should state that.

If the diagram is not an xml file a message should state that.

If the diagram was loaded no message should be shown.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
